service-runner.js

```diff
@@ -202,7 +250,15 @@ ServiceRunner.prototype._runMaster = function() {
     process.on('SIGTERM', shutdown_master);
 
     // Set up rolling restarts
-    process.on('SIGHUP', this._rollingRestart.bind(this));
+    process.on('SIGHUP', function() {
+        return self.updateConfig()
+        .then(function() {
+            // Recreate loggers
+            self._logger.close();
```

Comment:
`updateConfig` already sets up a new `logger` object, so this becomes useless really. Should be good once creating a new `logger` object is moved out of `updateConfig`